### PR TITLE
feat/ comment edit feature, comment fixes and js functionality

### PR DESCRIPTION
**Pull request by mikco-hipe**
### OVERVIEW

Added comment edit feature.
Restructure the Delete, Submit, Edit and Cancel button
Empty comments and comments exceeding 255 characters are not allowed.
Only owned comments can be edited.

I also fixed the JS functionality on displaying the image after selection on profile, post create and post update
### NOTES

added renaming migration file for renaming "content" to "comment"
"php artisan migrate" 

### SNIPPETS
Sample comment on post
Screenshot 2023-10-19 135204
editing of comment
Screenshot 2023-10-19 135223
edited comment 
Screenshot 2023-10-19 135233
rejects empty comment
Screenshot 2023-10-19 135248
rejects more than 255 characters
Screenshot 2023-10-19 135304
 the comment belongs to another user showing how they cannot edit other people's comment
Screenshot 2023-10-19 135554
database
image

### CODE REVIEW

**Review by hipe-francis — APPROVED** [review]
GEFM 🚀

File: laravel-app/app/Services/CommentService.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if (!$user) {
+            return redirect()->back()->with('error', 'You must be logged in to add a comment.');
+        }
+
+        $post->comments()->create([
+            'user_id' => $user->id,
+            'comment' => $request->comment,
+        ]);
+
+        return redirect()->back()->with('success', 'Comment added successfully.');
```
Comment (hipe01mark) [inline laravel-app/app/Services/CommentService.php:28]: Why does your service have redirections? what if I want to reuse the other controller and I don't want to have redirections to it??

File: laravel-app/app/Services/CommentService.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        $comment = PostComment::find($id);
+
+        if (!$comment) {
+            return redirect()->back()->with('error', 'Comment not found');
+        }
+
+        $comment->delete();
+
+        return redirect()->back()->with('success', 'Comment deleted successfully');
```
Comment (hipe01mark) [inline laravel-app/app/Services/CommentService.php:44]: Why does your service have redirections? what if I want to reuse the other controller and I don't want to have redirections to it??

File: laravel-app/app/Services/CommentService.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        $comment = PostComment::find($id);
+
+        if ($comment) {
+            $comment->comment = $validatedData;
+            $comment->save();
+            return redirect()->back()->with('success', 'Comment updated successfully.');
+        }
+
+        return redirect()->back()->with('error', 'Comment update failed.');
```
Comment (hipe01mark) [inline laravel-app/app/Services/CommentService.php:60]: Why does your service have redirections? what if I want to reuse the other controller and I don't want to have redirections to it??

File: laravel-app/app/Services/CommentService.php (hunk 1 above)
Comment (mikco-hipe) [reply laravel-app/app/Services/CommentService.php:28]: I'll transfer sir the redirection to the controller 🙇

File: laravel-app/app/Services/CommentService.php (hunk 3 above)
Comment (mikco-hipe) [reply laravel-app/app/Services/CommentService.php:60]: I'll transfer sir the redirection to the controller 🙇

File: laravel-app/app/Services/CommentService.php (hunk 2 above)
Comment (mikco-hipe) [reply laravel-app/app/Services/CommentService.php:44]: I'll transfer sir the redirection to the controller 🙇

**Review by hipe-francis — DISMISSED** [review]
GEFN 🚀

File: laravel-app/app/Services/CommentService.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+use App\Http\Requests\CommentRequest;
+use App\Models\PostComment;
+use App\Models\UserPost;
+
+class CommentService
+{
+    /**
+     * Logic for storing comment
+     */
+    public function storeComment(CommentRequest $request, UserPost $post): void
```
Comment (hipe01mark) [inline laravel-app/app/Services/CommentService.php:14]: Why does this method return void? What if I want to get the created comment and display it to my controller?

File: laravel-app/app/Services/CommentService.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * Logic for deleting comment
+     */
+    public function deleteComment(int $id): void
+    {
+        if ($comment = PostComment::find($id)) {
+            $comment->delete();
+        }
+    }
```
Comment (hipe01mark) [inline laravel-app/app/Services/CommentService.php:32]: Why does this method return void? What if I want to get the deleted comment and display it on my notification? Do I need to re-query again to get that and put it to my event handler??

File: laravel-app/app/Services/CommentService.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Logic for editing comment
+     */
+    public function editComment(int $id, string $validatedData): void
+    {
+        if ($comment = PostComment::find($id)) {
+            $comment->comment = $validatedData;
+            $comment->save();
+        }
+    }
```
Comment (hipe01mark) [inline laravel-app/app/Services/CommentService.php:43]: Why does this method return void? What if I want to get the updated comment and display it to my controller?

**Comment by hipe01mark** [comment]
Fix branch conflicts as well.

File: laravel-app/app/Services/CommentService.php (hunk 4 above)
Comment (mikco-hipe) [reply laravel-app/app/Services/CommentService.php:14]: Yes sir, will resolve it 🙇

File: laravel-app/app/Services/CommentService.php (hunk 5 above)
Comment (mikco-hipe) [reply laravel-app/app/Services/CommentService.php:32]: Yes sir, will resolve it 🙇

File: laravel-app/app/Services/CommentService.php (hunk 6 above)
Comment (mikco-hipe) [reply laravel-app/app/Services/CommentService.php:43]: Yes sir, will resolve it 🙇

**Review by bpocrafael — DISMISSED** [review]
GEFM

File: laravel-app/public/assets/js/app.js [hunk 7]
```diff
@@ -1,19 +1,40 @@
 // profile
-var profileImage = document.getElementById('profileImage');
-var fileInput = document.getElementById('profile_image');
+const profileImage = document.getElementById('profileImage');
+const profileFileInput = document.getElementById('profile_image');
 
-fileInput.addEventListener('change', function() {
-    if (fileInput.files.length > 0) {
-        var selectedFile = fileInput.files[0];
+if (profileFileInput){
```
Comment (hipe01mark) [inline laravel-app/public/assets/js/app.js:5]: check if state style format

**Review by hipe01mark — COMMENTED** [review]
LGTM :bow:

**Review by bpocrafael — APPROVED** [review]

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**